### ISSUE TITLE
Question: Is there a way around permission checks?

### ISSUE THREAD

**Issue by zblesk** [ArchiveBox/ArchiveBox#1247]
Because of space constraints, I had to move my big archive from the server to a NAS. I've then mounted the archive via NFS and pointed Archivebox to it.  
I haven't been able to get it running since. 

It always only gets to:
```
archivebox_1  | Change in ownership detected, please be patient while we chown existing files
archivebox_1  | This could take some time...
```

The archive is ~300GB big and because of how it's structured, all FS-level operations take a lot of time.

I've tried chowning it myself, which took over 4 hours. 
I apparently didn't hit the correct perm combo, because the message just appears again. 

I've let the container run for ~17 hours, which should be more than enough, but nothing changed. 
(I'm currently running a differenattempt at a maual fix, but I'm not holding out much hope.)

Is there any way around this? Overriding which user is used (UID and GID in Compose don'd seem to do anything), or maybe letting it run as root and skip perm checks, or something? 

(I have also temporarily stopped and disabled Sonic, just to be sure.)

Thanks.

**Comment by pirate** [comment]
I am thinking of removing these checks entirely. They were added to help beginner users with common permission issues when mounting via Docker Desktop on macOS/Windows, but they cause problems for advanced users on network filesystems or who want to manage permissions more granularly.

Until I release that change, as a temporary workaround, you can download and mount this script into docker and remove the permissions check lines here: https://github.com/ArchiveBox/ArchiveBox/blob/dev/bin/docker_entrypoint.sh#L16

This will override the entrypoint that runs when docker starts and should let you change it without needing to rebuild an entire custom docker image.

e.g.

`docker-compose.yml`:
```
version: '3.9'

services:
    archivebox:
        ...
        volumes:
            - ./data:/data
            - ./archivebox_sourcecode/bin/docker_entrypoint.sh:/app/archivebox/bin/docker_entrypoint.sh
```

**Comment by pirate** [comment]
Ok I removed the forced `chown` and replaced it with a warning telling the user how to fix it manually: https://github.com/ArchiveBox/ArchiveBox/commit/4f655fc4a1fc21759fe21f13b354d940ca53b4a2

**Comment by zblesk** [comment]
Great, thanks! Do you plan to push the change to Docker Hub anytime soon? 🙏🏻

**Comment by zblesk** [comment]
@pirate I've tried pulling the :dev image, and I'm seeing some weird behavior. 
All files were owned by my default user (1000:1000). But when I start the container, it starts changing the permissions - it sets ArchiveBox.conf to 999:999, then crashes with  `PermissionError: [Errno 13] Permission denied: '/data/ArchiveBox.conf'`. 
I've tried chowning it and starting the container again, but it just sets it to 999 again and crashes.

**Comment by pirate** [comment]
Hmm strange ok I'll remove the top-level chown. What OS are you using on the host?

**Comment by p0n1** [comment]
> @pirate I've tried pulling the :dev image, and I'm seeing some weird behavior. All files were owned by my default user (1000:1000). But when I start the container, it starts changing the permissions - it sets ArchiveBox.conf to 999:999, then crashes with `PermissionError: [Errno 13] Permission denied: '/data/ArchiveBox.conf'`. I've tried chowning it and starting the container again, but it just sets it to 999 again and crashes.

@pirate 

Similar issue here.

I used to use  image with `latest` tag and it works perfectly. I just switched to the latest `dev` tag for `SINGLEFILE_ARGS` feature today.

Got errors bellow:

```
2023-10-26T13:07:51.804235828Z touch: cannot touch '/data/archive/.permissions_test_safe_to_delete': Permission denied
2023-10-26T13:07:51.804394367Z /app/bin/docker_entrypoint.sh: line 25: 2: Permission denied
2023-10-26T13:07:51.804630917Z /app/bin/docker_entrypoint.sh: line 26: 2: Permission denied
2023-10-26T13:07:51.804882554Z /app/bin/docker_entrypoint.sh: line 27: 2: Permission denied
```

I just deployed a new instance for test. If I mount a empty folder, the `dev` version could start correctly. But it will fail to start if I just restarted the container and get permission errors like above.

I'm using a folder mounted by NFS.

**Comment by pirate** [comment]
@zblesk and @p0n1 both of your issues should be fixed by passing the PUID & PGID environment variables to the container containing the UID & GID you want the files to be owned by. This matches the behavior of all LinuxServer.io docker images, and is a general common standard for many containers. https://docs.linuxserver.io/general/understanding-puid-and-pgid

I also made some changes to the entrypoint script. Can you try setting those variables, pulling/building the latest `:dev` image, and running it again?

**Comment by mAAdhaTTah** [comment]
@pirate `dev` image isn't building, was last published [11 days ago](https://hub.docker.com/r/archivebox/archivebox/tags).

**Comment by p0n1** [comment]
Thanks for replying. @pirate I just deployed a new instance for testing again. I set the `PUID` and `GUID` as `999` which is the same as the host. I got the same error I posted before.

Just as @mAAdhaTTah raised, maybe I'm not using the latest `dev` image you mentioned.

**Comment by pirate** [comment]
My apologies, fixed the build and just pushed `dev`, (aka `0.7.0`, aka `latest`) for `amd64`, `arm64`, and `arm/v7` (all the 3 architectures ArchiveBox supports).

https://hub.docker.com/layers/archivebox/archivebox/dev/images/sha256-afa301d566a01cab8934aa1141b5f6133a4585ee5b449304c659c765e240575d?context=explore

**Comment by p0n1** [comment]
Thanks for the update @pirate. I just tested the latest `dev` tag with `PUID=1000` and `PGID=1000`. Everything works now.

I also figured out why many permission errors for my case and even can't start a new container instance when I mount the NFS folder filled with archivebox archives. 

It relates with the following code.

https://github.com/ArchiveBox/ArchiveBox/blob/c6e5a565c0b040358ab326d59f329c549e7f6ce0/bin/docker_entrypoint.sh#L21-L27

This script couldn't write into `/data` folder with `root` user for NFS folders. After I set `mapall` field in NFS share to the the original owner user in the host machine, this check will pass.

ref: `mapall` in https://man.freebsd.org/cgi/man.cgi?exports(5).
